tests/components/deluge/test_config_flow.py

```diff
@@ -0,0 +1,180 @@
+"""Test Deluge config flow."""
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.components.deluge.const import DEFAULT_NAME, DOMAIN
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_REAUTH, SOURCE_USER
+from homeassistant.const import CONF_SOURCE
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import (
+    RESULT_TYPE_ABORT,
+    RESULT_TYPE_CREATE_ENTRY,
+    RESULT_TYPE_FORM,
+)
+
+from . import CONF_DATA, IMPORT_DATA
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture(name="api")
+def mock_deluge_api():
+    """Mock an api."""
+    with patch("deluge_client.client.DelugeRPCClient.connect"), patch(
+        "deluge_client.client.DelugeRPCClient._create_socket"
+    ):
+        yield
+
+
+@pytest.fixture(name="auth_error")
+def mock_api_authentication_error():
+    """Mock an api."""
+    with patch(
+        "deluge_client.client.DelugeRPCClient.connect", side_effect=Exception
+    ) as client, patch("deluge_client.client.DelugeRPCClient._create_socket"):
+        client.side_effect.__name__ = "BadLoginError"
```

Comment:
`TypeError: can't set attributes of built-in/extension type 'Exception'`

This is an attempt on a workaround. Deluge uses python 2 so I do not know how to properly mock this.

Reply:
Still no solution for this yet.